src/nodes/utils/PostProcessingUtils.js

```diff
@@ -46,3 +48,42 @@ export const getScreenPosition = /*@__PURE__*/ Fn( ( [ viewPosition, projectionM
 	return vec2( sampleUv.x, sampleUv.y.oneMinus() );
 
 } );
+
+/**
+* Computes a normal vector based on depth data. Can be used as a fallback when no normal render
+* target is available or if flat surface normals are required.
+*
+* @param {vec2} uv - The texture coordinate.
+* @param {DepthTexture} depthTexture - The depth texture.
+* @param {mat4} projectionMatrixInverse - The camera's inverse projection matrix.
+* @return {vec3} The computed normal vector.
+*/
+export const getNormalFromDepth = /*@__PURE__*/ Fn( ( [ uv, depthTexture, projectionMatrixInverse ] ) => {
+
+	const size = textureSize( textureLoad( depthTexture ) );
+	const p = ivec2( uv.mul( size ) ).toVar();
+
+	const c0 = textureLoad( depthTexture, p ).toVar();
```

Comment:
@sunag It seems there is an issue when using depth textures with `textureLoad()` and the WebGL backend. There are no issues in WebGPU. The following shader errors occur:

> ERROR: 0:149: '=' : dimension mismatch
ERROR: 0:149: 'assign' : cannot convert from 'highp 4-component vector of float' to 'highp float'

Live example: https://jsfiddle.net/1tkpy5ja/

I think this is related to the fact that `texelFetch` always returns a four-component vector. `textureLoad` is more flexibel and can also return `f32` values. When I read the code correctly, this difference between GLSL and WGSL isn't honored yet.